zerver/actions/streams.py

```diff
@@ -201,6 +204,8 @@ def do_unarchive_stream(stream: Stream, new_name: str, *, acting_user: UserProfi
         ]
     )
 
+    ChannelEmailAddress.objects.filter(channel=stream).update(deactivated=False)
```

Comment:
Didn't add tests for this as `deactivated` field is not playing any practical role currently.

Reply:
Yeah that seems fine. We might want a comment above this line that this might be a decision we want to separate from channel reactivation in the future.

Reply:
> this might be a decision we want to separate from channel reactivation in the future.

I didn't understand this part. We'll always want these channel email addresses to be marked as deactivated when a channel is deactivated & reactivated when a channel is reactivated, right ?

Reply:
I think once they're a thing created more like an integration, we might want to leave them deactivated when a channel is reactivated.